### PR TITLE
DOP-3976: add a boost for phrase of the original query

### PR DESCRIPTION
**Pull request by seungpark**
### Ticket

DOP-3976

### Notes
- Adds a [phrase](https://mongodb.com/docs/atlas/atlas-search/phrase/) operator to search across documents' `paragraphs` `headings` and `text` for the original search phrase with standard analyzer. Added a boost to this score as we would want documents with these words in order to be higher in the ranking

### Staged changes:
Can see the change here:
https://mongodbcom-cdn.website.staging.corp.mongodb.com/docs-qa/search/?q=max+disk+iops&page=1

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [X] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

File: .drone.yml
Comment (mayaraman19) [inline .drone.yml:11]: Should this line be deleted?

File: src/Query/index.ts
Comment (mayaraman19) [inline src/Query/index.ts:190]: Should we add a `slop` value to this operator? Currently on the staging link, the results for, for example, "delete collection" and "collection delete" are quite different. Would it be worth adding a little bit of a `slop` value so that there's a bit of flexibility in how the query is asked?

File: tests/unit/Query.test.ts
Comment (mayaraman19) [inline tests/unit/Query.test.ts:36]: Could this test name be reworded? It took me a bit of time to understand what it means 😭 the "should" threw me off lol

File: .drone.yml
Comment (seungpark) [reply .drone.yml:11]: will delete before merge!

File: tests/unit/Query.test.ts
Comment (seungpark) [reply tests/unit/Query.test.ts:36]: good point! updated

File: src/Query/index.ts
Comment (seungpark) [reply src/Query/index.ts:190]: this can be a follow up but i think this will introduce more changes than the ask, in terms of ranking. this is specifically for the phrase asked without reordering.

but you bring up a good point - perhaps the boost is too large, as it will look for this phrase and boost for its existence - regardless of its content. will lower this boost while trying to keep the same result as described in the ticket

Comment (seungpark) [reply src/Query/index.ts:190]: @mayaraman19 added a slop factor for this and modified the boost! upon testing, this was pretty significant in boosting the phrase as is, and also preserving cases where order does not matter as in your example.

**Review by mmeigs — COMMENTED** [review]
This looks good. But I am wondering about the results of the search. Mainly how closely the phrase is found. I'm sure you've sent this to Sarah, but I did notice that the one result she said should be found by this search phrase is `https://www.mongodb.com/docs/atlas/review-available-metrics/` and that is currently on the second page of results, the 16th search result.

**Comment by seungpark** [comment]
> This looks good. But I am wondering about the results of the search. Mainly how closely the phrase is found. I'm sure you've sent this to Sarah, but I did notice that the one result she said should be found by this search phrase is `https://www.mongodb.com/docs/atlas/review-available-metrics/` and that is currently on the second page of results, the 16th search result.

good point. i think its safe to raise this boost factor since we added a slop factor. also left [a comment](https://jira.mongodb.org/browse/DOP-3976?focusedCommentId=6214136&focusedId=6214136&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-6214136) for possible difference in result on the issue

**Review by mayaraman19 — APPROVED** [review]
I think this looks good, I definitely see an improvement with the other examples in the ticket such as "delete collection" and "node.js driver" (and with slop factor, "collection delete" and "driver node.js"). Might want to get the results approved by Sarah as well when she's back?

**Review by sarahemlin — APPROVED** [review]
LGTM from Corry Root & me.  Thank you!

**Pull request merged**
